frontend/src/store/modules/sheet.ts

```diff
@@ -82,6 +94,55 @@ const getters = {
 
     return state.sheetById.get(sheetId) || (unknown("SHEET") as Sheet);
   },
+  isCreator: (
+    state: SheetState,
+    getters: any,
+    rootState: any,
+    rootGetters: any
+  ) => {
+    const currentUser = rootGetters["auth/currentUser"]();
+    const currentSheet = getters.currentSheet;
+
+    if (!currentSheet) return false;
+
+    return currentUser.id === currentSheet!.creator.id;
+  },
+  isReadOnly: (
+    state: SheetState,
+    getters: any,
+    rootState: any,
+    rootGetters: any
+  ) => {
+    const currentUser = rootGetters["auth/currentUser"]();
+    const { currentSheet } = getters;
+
+    if (!currentSheet) return true;
+    // creator always can edit
+    if (getters.isCreator) return false;
```

Comment:
These are usually bad patterns as it relies on internal state of the isCreator. Code like this is less readable since we need to check the isCreator implementation. Also it's less robust

e.g in a multi-thread environment, what if there is another thread changes the currentSheet after 117 and before 121?

The standard practice is try to use local state as much as possible. 

const currentSheet = getters.currentSheet;
isCreator(currentSheet)

Saw this couple times, OK to leave it for now in this PR. But after we finish the feature implementation, let's create a separate PR to change these.